File: Sources/App/DTOs/FanartTV/ArtistResponse.swift
//
//  ArtistResponse.swift
//  FinchServer
//
//  Created by Stefan Klein Nulent on 10/06/2025.
//

import Foundation

struct ArtistResponse: Decodable {
    
    // MARK: - Types
    
    enum CodingKeys: String, CodingKey {
        case name
        case mbId = "mbid_id"
        case backgrounds = "artistbackground"
        case images = "artistthumb"
        case musicLogos = "musiclogo"
        case hdMusicLogos = "hdmusiclogo"
    }
    
    
    
    // MARK: - Properties
    
    let name: String
    let mbId: String
    let backgrounds: [Image]?
    let images: [Image]?
    let musicLogos: [Image]?
    let hdMusicLogos: [Image]?
    
    var isImageAvailable: Bool {
        backgrounds?.isEmpty == false || images?.isEmpty == false || musicLogos?.isEmpty == false || hdMusicLogos?.isEmpty == false
    }
}

File: FinchServer/Metadata/FanartTV/DTO/ArtistResponse.cs
using System.Text.Json.Serialization;

namespace FinchServer.Metadata.FanartTV.DTO;

public class ArtistResponse {
    
    // - Types

    public class Image {
        
        [JsonPropertyName("url")]
        public required string Url { get; set; }
    }
    
    
    // - Properties
    
    [JsonPropertyName("name")]
    public required string Name { get; set; }
    
    [JsonPropertyName("mbid_id")]
    public required string MbId { get; set; }
    
    [JsonPropertyName("artistbackground")]
    public Image[]? Artistbackground { get; set; }
    
    [JsonPropertyName("artistthumb")]
    public Image[]? Artistthumb { get; set; }
    
    [JsonPropertyName("musiclogo")]
    public Image[]? Musiclogo { get; set; }
    
    [JsonPropertyName("hdmusiclogo")]
    public Image[]? Hdmusiclogo { get; set; }
}